sharktuner/sharktuner/libtuner.py

```diff
@@ -1191,6 +1201,27 @@ def compile(
     return compiled_candidates
 
 
+def filter_candidates_by_baseline(
+    candidate_results: list[BenchmarkResult],
+    baseline_handler: BaselineResultHandler,
+    num_candidates: int,
+    should_prune_slower_candidates: bool,
+) -> list[tuple[BenchmarkResult, float]]:
+    """
+    Filters and orders candidates based on baseline comparison.
+    """
+    if not baseline_handler.is_better_than_baseline(candidate_results):
+        if should_prune_slower_candidates:
+            return []
+
+    all_candidates_with_speedup = baseline_handler.get_candidates_ordered_by_speedup(
+        candidate_results
+    )
```

Comment:
Have you considered passing a second parameter to this function, `should_prune`? I'm wondering if we need a new function at all